barretenberg/cpp/src/barretenberg/plonk_honk_shared/composer/permutation_lib.hpp

```diff
@@ -45,40 +45,77 @@ struct cycle_node {
  *
  */
 struct permutation_subgroup_element {
-    uint32_t row_index = 0;
-    uint8_t column_index = 0;
+    uint32_t row_idx = 0;
+    uint8_t column_idx = 0;
     bool is_public_input = false;
     bool is_tag = false;
 };
 
+/**
+ * @brief Stores permutation mapping data for a single wire column
+ *
+ */
+struct Mapping {
```

Comment:
Do you think the new structure of 4 arrays is better than the original structure of a vector of structs? My thought is its probably slightly more efficient for memory, and therefore slightly better for cache efficiency as a result.

Reply:
The constructor went from 500ms to 30ms so thats all the proof I need. Having a vector/array of structs makes it impossible to use the slab allocator to allocate memory and difficult to multithread the initialization without either 0 initializing or doing lots of copying to "join" thread local data. I tried to find a slightly cleaner solution but this one was by far the fastest I could come up with

Reply:
got it, yeah. Is _allocate_aligned_memory multithreaded?